src/gen9_vdenc.c

```diff
@@ -3120,6 +3133,9 @@ gen9_vdenc_mfx_avc_inset_headers(VADriverContextP ctx,
 
     if (vdenc_context->is_frame_level_vdenc)
         insert_one_zero_byte = 0;
+    else {
+        insert_one_zero_byte = 1;
+    }
 
```

Comment:
The header for slice index zero makes sense.  I'm not so sure about whether multi data buffers are needed.